lib/marc/datafield.rb

```diff
@@ -33,38 +39,38 @@ class DataField
 
 
     # Create a new field with tag, indicators and subfields.
-    # Subfields are passed in as comma separated list of 
-    # MARC::Subfield objects, 
-    # 
+    # Subfields are passed in as comma separated list of
+    # MARC::Subfield objects,
+    #
     #   field = MARC::DataField.new('245','0','0',
     #     MARC::Subfield.new('a', 'Consilience :'),
     #     MARC::Subfield.new('b', 'the unity of knowledge '),
     #     MARC::Subfield.new('c', 'by Edward O. Wilson.'))
-    # 
+    #
     # or using a shorthand:
-    # 
+    #
     #   field = MARC::DataField.new('245','0','0',
     #     ['a', 'Consilience :'],['b','the unity of knowledge '],
     #     ['c', 'by Edward O. Wilson.'] )
 
     def initialize(tag, i1=' ', i2=' ', *subfields)
-      # if the tag is less than 3 characters long and 
+      # if the tag is less than 3 characters long and
       # the string is all numeric then we pad with zeros
       if tag.length < 3 and /^[0-9]*$/ =~ tag
         @tag = "%03d" % tag
       else
-        @tag = tag 
+        @tag = tag
       end
-      # can't allow nil to be passed in or else it'll 
+      # can't allow nil to be passed in or else it'll
       # screw us up later when we try to encode
       @indicator1 = i1 == nil ? ' ' : i1
       @indicator2 = i2 == nil ? ' ' : i2
-      
+
       @subfields = []
 
       # must use MARC::ControlField for tags < 010 or
       # those in MARC::ControlField#extra_control_fields
-      
+
       if MARC::ControlField.control_tag?(@tag)
```

Comment:
As above, test and raise exception prior to assignments.  If the validation should be a regex, then replace the constant `@@control_tags` with a regex in the `control_tag` method.  That is, the assignment in this method does a bit of manipulation to coerce it into a 3-digit string (if it's all digits), so perhaps a regex test is required?  In any case, whatever is required could be consolidated in `MARC::ControlField.control_tag?`